### PR TITLE
Control startup behavior based on launch configuration

### PR DESCRIPTION
**Pull request by msft-jlange**
Defines a new structure that encapsulates launch configuration into a configuration object and controls the behavior of the launch based on the encapsulated configuration

### CODE REVIEW

File: src/svsm.rs [hunk 1]
```diff
@@ -449,31 +449,43 @@ pub extern "C" fn svsm_main() {
...
-    let fw_meta = map_and_parse_fw_meta()
-        .unwrap_or_else(|e| panic!("Failed to parse FW SEV meta-data: {:#?}", e));
+    let fw_metadata: Option<SevFWMetaData>;
+    if config.should_launch_fw() {
+        fw_metadata = Some(
+            map_and_parse_fw_meta()
+            .unwrap_or_else(|e| panic!("Failed to parse FW SEV meta-data: {:#?}", e)));
+    } else {
+        fw_metadata = None;
+    }
```
Comment (00xc) [inline src/svsm.rs:459]: The `fw_metadata` can be constructed directly:

```suggestion
    let fw_metadata = if config.should_launch_fw() {
        Some(
            map_and_parse_fw_meta()
                .unwrap_or_else(|e| panic!("Failed to parse FW SEV meta-data: {:#?}", e)),
        )
    } else {
        None
    };
```

Or even better:

```suggestion
    let fw_metadata = config.should_launch_fw().then(|| {
        map_and_parse_fw_meta()
            .unwrap_or_else(|e| panic!("Failed to parse FW SEV meta-data: {:#?}", e))
    });
```

**Review by 00xc — COMMENTED** [review]
Overall looks good to me, just left one suggestion below. Also, please sign your commits via `git commit -s` and run your changes through `cargo fmt` and `cargo clippy`. CI should have warned you about these but it's disabled because it's your first contribution.

**Comment by msft-jlange** [comment]
Thanks for the feedback!  I’m going to go with the first of your two proposed options, since the essence of any good code is readability, and the second option uses esoteric Rust syntax that I personally find much less readable (plus the second option will be much more difficult to integrate with future changes that integrate a different option provider).

-Jon

**Review by 00xc — COMMENTED** [review]
I'm still seeing some clippy warnings (`cargo clippy`). Would you mind fixing those as well? Thanks!

**Comment by msft-jlange** [comment]
Strange - I did run cargo clippy and it didn't give me any errors.  And now that you've prompted me to run it again, it did generate warnings.  I've fixed them and updated the PR.  Thanks for being patient with me!

-Jon

**Pull request merged**